src/java/org/apache/cassandra/hints/HintsServiceMBean.java

```diff
@@ -40,4 +42,11 @@
      * being dispatched right now, or being written to).
      */
     void deleteAllHintsForEndpoint(String address);
+
+    /**
+     * Lists all the endpoints that this node has hints for.
+     *
+     * @return a list of endpoints with relevant hint information - total number of files, newest and oldest timestamps.
+     */
+    List<PendingHintsInfo> listPendingHints();
```

Comment:
probably I'd rename it to `getPendingHintsInfo` or `getAllPendingHints`. WDYT?

the javadoc is not correct anymore, right? I'd say "Returns all pending hints that this node has."

the same changes need to be addressed in `HintsService`.

Reply:
Changed in https://github.com/apache/cassandra/pull/1232/commits/8787a817ca6b646cf7be10d70f65e0beb5be69bf.